var/spack/repos/builtin/packages/chexmix/package.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class Chexmix(Package):
+    """ChExMix: the ChIP-exo mixture model. ChExMix aims to characterize protein-DNA binding
+    subtypes in ChIP-exo experiments."""
+
+    homepage = "http://mahonylab.org/software/chexmix"
+    url = "https://github.com/seqcode/chexmix/releases/download/v0.52/chexmix.v0.52.public.jar"
+
+    version(
+        "0.52.public",
```

Comment:
One question before merging: why using the `public` suffix? Can we move the suffix to the package name?

Reply:
Although the [tags](https://github.com/seqcode/chexmix/tags) don't include `public`, the `.jar` files are either generated with it (`0.52`, `0.41`) or without (`0.51`) without any obvious reason ...

Reply:
If there are no differences, let's just use:
```suggestion
        "0.52",
```
?

Reply:
Can do, but then we'd need to specify a manual `url=` for `0.52` and any other versions for which the download contains `public`.
```
# spack versions chexmix
==> Safe versions (already checksummed):
  0.52.public
==> Remote versions (not yet checksummed):
  0.51  0.41.public
# spack checksum chexmix 0.52
==> Error: Could not find any remote versions for chexmix
```

Reply:
Let's give https://github.com/seqcode/chexmix/issues/6 some time. If there's a reply we might solve both issues.

Reply:
Just looping back to some of my older PRs, @alalazo.

Silence there for now, so what's your preference here - leave as is (version searches work, but some will be end up being `.public` suffixed) or manually specify urls for the different versions?

Reply:
Manually specified urls: more maintenance work, but also saner for end users

Reply:
Have switched over to manual urls. Bit more fiddly than just that as we also need control over the name of the downloaded file for the `install()` process, so have added a `run_before()` decorator to handle that.